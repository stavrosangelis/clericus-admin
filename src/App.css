body {
  font-family: 'Roboto',"Helvetica Neue",Arial,sans-serif;
  font-size: 15px;
}
.logo-mini {
  padding: 12px 0 10px 5px!important;
  margin: 0 !important;
}
.logo-normal {
  padding: 11px 0 0 !important;
}
.logo-container {
  position: relative;
  width: 20px;
  height: 20px;
  display: block;
}
.triangle-left {
  position: absolute;
  bottom: 0;
  left: 0;
  width: 0;
  height: 0;
  border-bottom: 20px solid #fff;
  border-right: 20px solid transparent;
  z-index: 1;
}
.triangle-left-inner {
  position: absolute;
  bottom: 2px;
  left: 2px;
  width: 0;
  height: 0;
  border-bottom: 13px solid #212120;
  border-right: 13px solid transparent;
  z-index: 2;
}
.triangle-right {
  position: absolute;
  bottom: 0;
  left: 0;
  width: 0;
  height: 0;
  border-bottom: 20px solid #fff;
  border-left: 20px solid transparent;
  z-index: 3;
}
.triangle-right-inner {
  position: absolute;
  bottom: 2px;
  left: 5px;
  width: 0;
  height: 0;
  border-bottom: 13px solid #212120;
  border-left: 13px solid transparent;
  z-index: 4;
}
.sidebar .logo:after, .off-canvas-sidebar .logo:after {
  width: 100%;
  left: 0;
  right: 0;
  background-color: rgba(255,255,255,0.2);
}
.sidebar .nav li > a, .off-canvas-sidebar .nav li > a {
  margin: 0 15px 0;
}
.nav-link {
  padding-top: 14px;
  padding-bottom: 14px;
}
.nav-link.active {
  background-color: rgba(255,255,255, 0.3);
  border-radius: 2px;
}
.sidebar .nav li > a, .sidebar .nav li > a.active, .sidebar .nav li > a:visited {
  color: #fff !important;
  opacity: 0.86;
  padding: 10px 0;
}
.sidebar .nav i {
  color: #fff !important;
  font-size: 28px;
}
.sidebar .nav p {
  font-weight: 600;
}
.sidebar .nav li {
  width: calc(100% - 1px)
}
.sidebar .nav li:hover {
  background-color: #1F272B;
}
.sidebar:after,
.off-canvas-sidebar:after {
  background: #242d31 !important;
}
.main-navbar {
  background-color: #fff;
  box-shadow: 0 1px 3px rgba(0,0,0,0.1);
  color: #333 !important;
}
.main-navbar a {
  color: #333 !important;
}
.main-footer {
  background-color: #fff;
  box-shadow: 0 -1px 3px rgba(0,0,0,0.1);
}
.main-panel > .content {
  margin-top: 20px;
}
.breadcrumb {
  background-color: #ddd;
  justify-content: flex-end;
}
.form-group label {
  display: block;
  font-weight: 600;
  color: #555;
  font-size: 15px;
}

h1 {
  font-size: 32px;
  padding: 5pt 0;
}
h2 {
  font-size: 28px;
  padding: 5pt 0;
}
h3 {
  font-size: 22px;
  padding: 5pt 0;
}
h4 {
  font-size: 18px;
  padding: 5pt 0;
  margin: 0;
}
header {
  margin-bottom: 60px;
}

/* multi level menu */
.sidebar .nav i {
  margin-top: -3px;
  font-size: 24px;
}
.sidebar a {
  color: #fff;
  opacity: 0.7;
}
.sidebar a:hover {
  opacity: 1;
}
.treeview-menu {
  font-size: 14px;
  width: calc(100% - 2px);
  background-color: #333;
  visibility: hidden;
  padding: 0;
  opacity: 0;
  height:0;
  transition: visibility 0s, opacity 0.2s linear;
  overflow-y: hidden;
}
.treeview-menu i {
  margin-right: 0 !important;
  font-size: 15px !important;
}
.treeview-menu.active {
  visibility: visible;
  height: auto;
  opacity: 1;
}
.toggle-treeview-icon {
  margin-top: -3px;
  margin-right: 0 !important;
  float: right !important;
  font-size: 18px !important;
  transition: transform 0.2s;
}
.toggle-treeview-icon.active {
  transform: rotate(-90deg);
}
.treeview-menu .nav-link {
  padding: 7px 5px 7px 25px;
}
.treeview-menu .nav-link.active {
  opacity: 1;
  background-color: #555;
  border-radius: 0;
  border-left: 4px solid #508CB7;
}
.class-piece-thumbnail {
  max-width: 160px;
  max-height: 160px;
}
.classpiece-wrapper {
  display: block;
  position: relative;
  overflow: scroll;
  width: 100%;
  height: calc(100vh - 160px);
}
.classpiece-container {
  top: 0;
  left: 0;
  display: block;
  position: relative;
  margin: 0;
  padding: 0;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -khtml-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}
.classpiece-cover {
  position: relative;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: #fff;
  z-index: 99991;
  width: 100%;
  height: 100vh
}
.classpiece-container img {
  position: absolute;
  top: 0;
  left: 0;
}
.classpiece-actions > div {
  display: block;
  margin-bottom: 10pt;
}
.face-boxes-container, .linking-boxes-container {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index:9999;
}
.text-boxes-container, .linking-text-container {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index:999;
}
.facebox, .textbox {
  background-color: transparent;
  position: absolute !important;
  z-index:99;
}
.facebox {
  border: 4pt solid #13dbca;
}
.textbox {
  outline: 4pt solid #00ffa1;
}
.facebox.has-thumbnail {
  background-color: rgba(52,219,58,0.3);
}
.facebox.active, .textbox.active {
  background-color: rgba(185,247,42,0.3);
}
.facebox.active.has-data,  .facebox.has-data{
  background-color: rgba(52,219,58,0.3);
}
.selected-face-thumbnail {
  max-height: 180px;
}

.pull-left {
  float: left;
}
.pull-right {
  float: right;
}
.box-tools {
  padding: 15px 0;
}


/* zoom slider */
.slidecontainer {
  width: 100%;
  display: block;
}

.slider {
  -webkit-appearance: none;
  appearance: none;
  width: 100%;
  height: 15px;
  border-radius: 15px;
  border:1px solid #d3d3d3;
  outline: none;
  opacity: 0.7;
  -webkit-transition: .2s;
  transition: opacity .2s;
}
.slider:hover {
  opacity: 1;
}

.slider::-webkit-slider-thumb:before {
  content: "\f002";
}
.slider::-webkit-slider-thumb {
  -webkit-appearance: none;
  appearance: none;
  width: 25px;
  height: 25px;
  border-radius: 100%;
  background: #4CAF50;
  cursor: pointer;
}

.slider::-moz-range-thumb {
  width: 25px;
  height: 25px;
  background: #4CAF50;
  cursor: pointer;
}
.toggle-icon {
  font-size: 17pt;
  cursor: pointer;
}
.toggle-icon.fa-toggle-on {
  color: #18bc6d;
}

.toggle-icon.fa-toggle-off {
  color: #e01a63;
}
.nav-block {
  border-bottom: 1px solid #ddd;
  padding: 10px 0;
}
.nav-block label {
  color: #999;
}
.nav-block:last-child {
  border-bottom: none;
}
.nav-block-body {
  display: block;
  overflow-y: hidden;
  transition: max-height 0.3s;
  max-height: 0;
}
.nav-block-body.active {
  max-height: 200px;
}
.toolbox-tooltip {
  z-index: 999999;
}
.tooltip > .tooltip-inner {
  background-color: rgba(81,100,119, 0.95);
  text-align: left;
}
.tooltip > .tooltip-inner > ul {
  margin: 0 0 0 15px;
  padding: 0;
}

.linking-selection {
  outline: 5pt #4CAF50 dashed;
  position: absolute;
  z-index: 99999999;
  display: none;
}
.linking-selection.active {
  display: block;
}
.thumbnail-textbox {
  background-color: #4CAF50;
  display: inline-block;
  margin: 2px 5px;
  padding: 2px 5px;
  color: #FFF;
  cursor: move;
  border: 1px solid #41a045;
  border-radius: 2px;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -khtml-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.context-menu {
  position: fixed;
  z-index: 999999;
  background-color: #fff;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  width: 180px;
  display: none;
}
.context-menu .header {
  font-style: italic;
  font-size: 13px;
  color: #666;
  padding: 5px;
}
.context-menu ul {
  padding: 0;
  margin: 0;
  list-style-type: none;
}
.context-menu ul li {
  border: 1px solid #fff;
  position: relative;
  display: block;
  padding: 4px 20px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}
.context-menu ul li:hover {
  border: 1px solid #ddd;
  background-color: #ddd;
}
.context-menu ul li.disabled, .context-menu ul li:hover.disabled {
  border: 1px solid #fff;
  font-style: italic;
  color: #666;
  background-color: #fff;
}
.context-menu .separator {
  border-top: 1px solid #ccc;
  height: 0;
  padding: 0;
  margin: 0;
}
.context-menu ul li i {
  position: absolute;
  right: 3px;
  top: 7px;
}
.context-menu ul li > ul {
  display: none;
  position: absolute;
  left: 180px;
  top: 0;
  background-color: #fff;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  width: 180px;
}
.context-menu ul li:hover > ul {
  display: block;
}
.hidden {
  display: none !important;
}

.import-to-db-table {
  width: 100%;
  border: 1px solid #eee;
}
.import-to-db-table td {
  vertical-align: top;
  padding: 5px;
  font-size: 14px;
}
.import-to-db-thumb {
  max-width: 100px;
  max-height: 150px;
}
.import-to-db-thumb-small {
  max-width: 20px;
  max-height: 40px;
}
.import-to-db-table tr td:first-child {
  vertical-align: middle;
}
.import-to-db-table tr:nth-child(odd) td {
  background-color: #fff;
}
.import-to-db-table tr:nth-child(even) td {
  background-color: #f9f9f9;
}
.import-to-db-table tr:hover td {
  background-color: #eee;
}
.import-to-db-table tr:last-child td {
  border: none;
}
.import-to-db-table label {
  font-size: 16px;
  font-weight: 600;
  display: block;
}


.select-checkbox-container {
  display: block;
  position: relative;
  margin: -13px 0 0 0;
  padding: 0;
  cursor: pointer;
  font-size: 22px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 100%;
}
.select-checkbox-container input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
}
.select-checkbox-container input ~ .select-checkbox {
  border-radius: 50%;
}
.select-checkbox-container input:checked ~ .select-checkbox {
  background: #2196F3 !important;
}
.select-checkbox {
  position: absolute;
  top: 0;
  left: 0;
  height: 25px;
  width: 25px;
  background: radial-gradient(#ccc, #ddd);
}
.select-checkbox:after {
  content: "";
  position: absolute;
  display: none;
}
.select-checkbox-container input:checked ~ .select-checkbox:after {
  display: block;
}
.select-checkbox-container .select-checkbox:after {
  left: 9px;
  top: 6px;
  width: 5px;
  height: 10px;
  border: solid white;
  border-width: 0 3px 3px 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}
.collapse-trigger {
  color: #3f8da3;
  font-weight: 500;
  cursor: pointer;
}
.collapse-trigger:hover {
  color: #34a3c1;
}
.collapse-trigger:active {
  color: #3f8da3;
}
.collapse-trigger i {
  font-size: 17px;
  line-height: 16px;
  vertical-align: middle;
  margin: -2px 0 0 5px;
  position: relative;
  transition: transform 0.2s;
}
.collapse-trigger i.active {
  transform: rotate(-90deg);
}
.collapse-relations-list {
  box-shadow: -2px 0 10px rgba(0,0,0,0.1) inset;
  list-style-type: none;
  padding:15px;
  border: 1px solid #ddd;
  max-height: 300px;
  overflow-y:auto;
}
.collapse-relations-list li::before {
  content: "-";
  padding: 0 10px 0 0;
  margin: 0 0 0 -10px;
}
.collapse-relations-list li {
  padding: 5px 0;
}
.collapse-relations-list li:hover {
  background-color: #f9f9f9;
}
.collapse-relations-list li > .relation {
  font-style: italic;
}
.collapse-relations-list li > .type-of {
  color: #666;
}
.import-to-db-details {
  display: inline-block;
  vertical-align: top;
  padding-right: 15px;
}
.import-to-db-details.duplicates {
  border-right: 1px dashed #ccc;
}
.import-to-db-duplicates {
  display: inline-block;
  vertical-align: top;
  padding-left: 15px;
  border-left: 1px dashed #ccc;
  margin-left: -1px;
}
.import-to-db-duplicates label {
  font-size: 15px;
  font-weight: normal;
  font-style: italic;
  margin: 0;
}
.import-to-db-duplicates ul {
  padding: 0;
  list-style-type: none;
}
.import-to-db-duplicates ul li {
  padding: 2px 0;
}
@media screen and (min-width: 992px) {
  .navbar .navbar-toggle {
      display: block !important;
  }
  .navbar .navbar-toggler-bar {
    display: block;
    position: relative;
    width: 22px;
    height: 1px;
    border-radius: 1px;
    background: #66615B;
  }
  .navbar-expand-lg .navbar-toggler.left {
    display: block !important;
  }
  .bar1, .bar2, .bar3 {
    outline: 1px solid transparent;
  }
  .bar1 {
    top: 0px;
    -webkit-animation: topbar-back 500ms linear 0s;
    animation: topbar-back 500ms 0s;
    -webkit-animation-fill-mode: forwards;
    animation-fill-mode: forwards;
  }
  .navbar .navbar-toggler-bar.bar2 {
    width: 17px;
    transition: width .2s linear;
  }
  .navbar .navbar-toggler-bar + .navbar-toggler-bar {
    margin-top: 7px;
  }
  .bar3 {
    bottom: 0px;
    -webkit-animation: bottombar-back 500ms linear 0s;
    animation: bottombar-back 500ms 0s;
    -webkit-animation-fill-mode: forwards;
    animation-fill-mode: forwards;
  }

  /* sidebar toggle */
  .nav-open .sidebar {
    -webkit-transform: translate3d(0px, 0, 0);
    transform: translate3d(0px, 0, 0);
  }
  .sidebar, .bootstrap-navbar {
    position: fixed;
    display: block;
    top: 0;
    height: 100%;
    width: 260px;
    right: auto;
    left: 0;
    z-index: 1032;
    visibility: visible;
    overflow-y: visible;
    padding: 0;
    transition: all 0.5s cubic-bezier(0.685, 0.0473, 0.346, 1);
    -webkit-transform: translate3d(-260px, 0, 0);
    transform: translate3d(-260px, 0, 0);
  }
  .nav-open .main-panel {
    position: relative;
    float: right;
    width: calc(100% - 260px);
    background-color: #f4f3ef;
    transition: all 0.5s cubic-bezier(0.685, 0.0473, 0.346, 1);
  }
  .main-panel {
    width: 100%;
  }

}
@media screen and (max-width: 991px) {
  .nav-open .main-panel {
    right: 0;
    -webkit-transform: translate3d(260px, 0, 0);
    transform: translate3d(260px, 0, 0);
    width: calc(100% - 260px);
    float: left;
  }
}
.fixed-plugin .fa-cog {
  cursor: pointer;
}
.context-menu .header {
  margin-bottom: 0;
}
.duplicate-modal h4 {
  padding-top: 0;
  margin-top: 0;
}
.duplicate-thumbnail-placeholder {
  min-width: 100px;
  max-width: auto;
  min-height: 100px;
  max-height: auto;
  display: inline-block;
  border-radius: 0.25rem;
  background-color: #ccc;
}
.duplicate-thumbnail-placeholder.over {
  outline: 3px dashed rgba(100,173,229, 0.7);
}
.thumbnail-person {
  cursor: pointer;
  display: block;
  position: relative;
  overflow: hidden;
}
.thumbnail-person-edit {
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  height: 0;
  transition: height 0.2s;
  background-color: rgba(100,173,229, 0.9);
  color: #fff;
  z-index: 9999;
  text-align: center;
  line-height: 26px;
}
.thumbnail-person img {
  z-index: 99;
}
.thumbnail-person:hover .thumbnail-person-edit {
  height: 26px;
}

.breadcrumb {
  font-size: 13px;
  padding: 0.5rem;
}
.breadcrumb-item a {
  color: #137184;
}
.breadcrumb-item a:hover {
  text-decoration: none;
  color: #1689a0;
}
.breadcrumb-item a:active {
  color: #105b6b;
}
.card {
  border-radius: 4px !important;
}
.first-cap {
  text-transform: capitalize;
}
.dropdown-item.active {
  background-color: #b7a380;
}
.page-actions {
  padding-bottom: 15px;
  display: block;
  position: relative;
  min-height: 40px;
}
.metadata-title {
  font-weight: 500;
  text-transform: capitalize;
}
.list-items {
  list-style-type: none;
  margin: 0 0 0 15px;
  padding: 0;
}
.btn-xs {
  padding: 1px 5px;
  line-height: 1;
}

.collapse-toggle {
  padding-right: 3px;
  transition: transform 0.2s;
}
.collapse-toggle.active {
  transform: rotate(-90deg);
}
.related-resource {
  max-width: 120px;
  display: inline-block;
  margin: 5px 10px;
}
