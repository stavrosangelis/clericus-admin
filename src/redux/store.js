import {createStore, applyMiddleware} from "redux";
import rootReducer from "./reducers/index";
import thunk from 'redux-thunk';

export const defaultState = {
  // settings
  settings: {},
  seedRedirect: false,

  // languageCodes
  languageCodes: [],

  // login
  loginError: false,
  loginErrorText: [],
  sessionActive: false,
  sessionUser: null,
  loginRedirect: false,

  resourcesPagination: {
    limit:25,
    activeType:null,
    page:1,
    orderField: "label",
    orderDesc: false,
    status: null,
    searchInput: "",
  },
  peoplePagination: {
    limit:25,
    peopleType:null,
    page:1,
    orderField: "firstName",
    orderDesc: false,
    status: null,
    searchInput: "",
    advancedSearchInputs: []
  },
  organisationsPagination: {
    limit:25,
    page:1,
    orderField: "label",
    orderDesc: false,
    activeType: null,
    status: null,
    searchInput: "",
  },
  eventsPagination: {
    limit:25,
    page:1,
    orderField: "label",
    orderDesc: false,
    activeType: null,
    status: null,
    searchInput: "",
  },
  temporalsPagination: {
    limit:25,
    page:1,
    orderField: "label",
    orderDesc: false,
    status: null,
    searchInput: "",
  },
  spatialsPagination: {
    limit:25,
    page:1,
    orderField: "label",
    orderDesc: false,
    status: null,
    searchInput: "",
  },
  usersPagination: {
    limit:25,
    page:1,
    orderField: "firstName",
    orderDesc: false,
    status: null,
    searchInput: "",
  },
  usergroupsPagination: {
    limit:25,
    page:1,
    orderField: "label",
    orderDesc: false,
    status: null,
    searchInput: "",
  },
  articlesPagination: {
    limit:25,
    page:1,
    activeType: null,
    orderField: "label",
    orderDesc: false,
    status: null,
    searchInput: "",
  },
  slideshowPagination: {
    limit:25,
    page:1,
    orderField: "order",
    orderDesc: false,
    status: null,
    searchInput: "",
  },
  contactFormsPagination: {
    limit:25,
    page:1,
    orderField: "createdAt",
    orderDesc: true,
    searchInput: "",
  },
  userGroups: [],

  resourcesTypes: [],
  personTypes: [],
  organisationTypes: [],
  eventTypes: [],

  lightBoxOpen: false,
  lightBoxSrc: null,

  peopleRoles: [],

  // entities properties
  entitiesLoaded:false,
  eventEntity: null,
  organisationEntity: null,
  personEntity: null,
  resourceEntity: null,
  temporalEntity: null,
  spatialEntity: null,
}

function configureStore(state = defaultState) {
  const store = createStore(
    rootReducer,
    state,
    applyMiddleware(thunk),
  );
  return store;
}
export default configureStore;
