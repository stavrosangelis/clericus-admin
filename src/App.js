import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Switch , Redirect} from 'react-router-dom';

import 'axios-progress-bar/dist/nprogress.css';
import 'bootstrap/dist/css/bootstrap.css';
import "./assets/scss/paper-dashboard.css";
import "./assets/demo/demo.css";
import './assets/fonts/roboto/css/roboto.css';
import './assets/fonts/font-awesome/css/font-awesome.min.css';
import "./assets/fonts/pe-icon-7/css/pe-icon-7-stroke.css";
import '../node_modules/leaflet/dist/leaflet.css';
import './assets/react-vis/style.css';
import './App.css';
import {loadProgressBar} from 'axios-progress-bar';

// routes
import indexRoutes from "./routes/index";
import Login from "./views/login";
import Seed from "./views/seed";

// layout components
import Header from "./components/header";
import Footer from "./components/footer";
import Sidebar from "./components/sidebar";
import {connect} from "react-redux";
import {
  getSystemTypes,
  getPeopleRoles,
  getPersonTypes,
  getOrganisationTypes,
  getEventTypes,
  loadDefaultEntities,
  loadSettings,
  checkSession,
  resetLoginRedirect,
  resetSeedRedirect,
  getLanguageCodes
} from "./redux/actions/main-actions";

// set default axios token;
import axios from 'axios';
let authToken = localStorage.getItem('token');
if (typeof authToken!=="undefined" && authToken!==null) {
  axios.defaults.headers.common['Authorization'] = 'Bearer '+authToken;
}

const mapStateToProps = state => {
  return {
    sessionActive: state.sessionActive,
    loginRedirect: state.loginRedirect,
    seedRedirect: state.seedRedirect,
    settings: state.settings
   };
};

function mapDispatchToProps(dispatch) {
  return {
    loadSettings: () => dispatch(loadSettings()),
    getSystemTypes: () => dispatch(getSystemTypes()),
    getPeopleRoles: () => dispatch(getPeopleRoles()),
    getPersonTypes: () => dispatch(getPersonTypes()),
    getOrganisationTypes: () => dispatch(getOrganisationTypes()),
    getEventTypes: () => dispatch(getEventTypes()),
    loadDefaultEntities: () => dispatch(loadDefaultEntities()),
    checkSession: () => dispatch(checkSession()),
    resetLoginRedirect: ()=>dispatch(resetLoginRedirect()),
    resetSeedRedirect: ()=>dispatch(resetSeedRedirect()),
    getLanguageCodes: ()=>dispatch(getLanguageCodes()),
  }
}

class App extends Component {
  constructor(props){
    super(props);
    this.state = {
      backgroundColor: "black",
      activeColor: "info",
      loginRedirect: false
    }

    this.openSidebar = this.openSidebar.bind(this);
    this.parseRoutes = this.parseRoutes.bind(this);
    this.parseChildrenRoutes = this.parseChildrenRoutes.bind(this);
  }

  openSidebar() {
    if (window.innerWidth > 992) {
      document.documentElement.classList.toggle("nav-open");
    }
  }

  parseRoutes() {
    let routes = [];
    for (let i=0; i<indexRoutes.length; i++) {
      let route = indexRoutes[i];
      let newRoute=[];
      if (route.component!==null) {
        newRoute = <Route path={route.path} key={i} component={route.component} />;
      }
      if (route.name==="Home") {
        newRoute = <Route exact path={route.path} key={i} component={route.component} />;
      }
      routes.push(newRoute);
      if (typeof route.children!=="undefined" && route.children.length>0) {
        let childrenRoutes = this.parseChildrenRoutes(route.children, i);
        routes.push(childrenRoutes);
      }
    }
    let flattenedRoutes = [].concat.apply([], routes);
    return flattenedRoutes;
  }

  parseChildrenRoutes(children, i) {
    let routes = [];
    for (let j=0;j<children.length; j++) {
      let childRoute = children[j];
      routes.push(<Route path={childRoute.path} key={i+"."+j} component={childRoute.component} />);
      if (typeof childRoute.children!=="undefined" && childRoute.children.length>0) {
        let childrenCRoutes = this.parseChildrenRoutes(childRoute.children,i+"."+j);
        routes.push(childrenCRoutes);
      }
    }
    return routes;
  }

  componentDidMount() {
    this.props.loadSettings();
    this.props.checkSession();
    this.props.getLanguageCodes();
  }

  componentDidUpdate(prevProps) {
    if (!prevProps.sessionActive && this.props.sessionActive) {
      this.openSidebar();
      this.props.getSystemTypes();
      this.props.getPeopleRoles();
      this.props.getPersonTypes();
      this.props.getOrganisationTypes();
      this.props.getEventTypes();
      this.props.loadDefaultEntities();
      loadProgressBar();
    }
    if (this.props.loginRedirect) {
      this.props.resetLoginRedirect();
    }
    if (this.props.seedRedirect) {
      this.props.resetSeedRedirect();
    }
  }

  render() {
    let settings = this.props.settings;
    let routes = this.parseRoutes();
    let loginRedirect = [];
    let seedRedirect = [];

    if (this.props.loginRedirect) {
      loginRedirect = <Redirect to='/' />
    }
    if (this.props.seedRedirect) {
      seedRedirect = <Redirect to='/' />
    }
    let AppHTML = [];
    if (typeof settings.seedingAllowed!=="undefined" && settings.seedingAllowed) {
      AppHTML = <Router basename={process.env.REACT_APP_BASENAME}>
        {seedRedirect}
        <Seed />
      </Router>
    }
    else if (!this.props.sessionActive) {
      AppHTML = <Login />
    }
    // note to handle session active but no login
    else if (this.props.sessionActive) {
      AppHTML = <Router basename={process.env.REACT_APP_BASENAME}>
        <div className="wrapper" id="main-wrapper">
          {loginRedirect}
          <Sidebar
            {...this.props}
            routes={indexRoutes}
            bgColor={this.state.backgroundColor}
            activeColor={this.state.activeColor}
          />
          <div className="main-panel" id="main-panel">
            <Header {...this.props} />
            <div className="content" id="content-container">
              <Switch>
                {routes}
              </Switch>
            </div>
            <Footer fluid extraClass="main-footer"/>
          </div>
        </div>
      </Router>
    }
    return (
      <div>
        {AppHTML}
      </div>
    );
  }
}

export default App = connect(mapStateToProps, mapDispatchToProps)(App);
