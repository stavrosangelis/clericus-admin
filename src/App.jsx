import React, { useCallback, useEffect, useState } from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect,
} from 'react-router-dom';
import { loadProgressBar } from 'axios-progress-bar';
import { useDispatch, useSelector } from 'react-redux';
import axios from 'axios';

// css
import 'axios-progress-bar/dist/nprogress.css';
import 'bootstrap/dist/css/bootstrap.css';
import './assets/fonts/roboto/css/roboto.css';
import './assets/fonts/font-awesome/css/font-awesome.min.css';
import './assets/fonts/pe-icon-7/css/pe-icon-7-stroke.css';
import '../node_modules/leaflet/dist/leaflet.css';
import './assets/scss/App.scss';

// routes
import indexRoutes from './routes/index';
import Login from './views/login';
import Seed from './views/seed';

// layout components
import Header from './components/header';
import Footer from './components/footer';
import Sidebar from './components/sidebar';

import {
  getSystemTypes,
  getPeopleRoles,
  getPersonTypes,
  getOrganisationTypes,
  getEventTypes,
  loadDefaultEntities,
  loadSettings,
  checkSession,
  resetLoginRedirect,
  resetSeedRedirect,
  getLanguageCodes,
} from './redux/actions';

const APIPath = process.env.REACT_APP_APIPATH;
const basename = process.env.REACT_APP_BASENAME;
// set default axios token;
const authToken = localStorage.getItem('token');
if (typeof authToken !== 'undefined' && authToken !== null) {
  axios.defaults.headers.common.Authorization = `Bearer ${authToken}`;
}

function App() {
  // redux store
  const dispatch = useDispatch();
  const sessionActive = useSelector((state) => state.sessionActive);
  const loginRedirect = useSelector((state) => state.loginRedirect);
  const seedRedirect = useSelector((state) => state.seedRedirect);
  const settings = useSelector((state) => state.settings);

  // state
  const [sessionRedirect, setSessionRedirect] = useState(false);

  const backgroundColor = 'black';
  const activeColor = 'info';

  const openSidebar = () => {
    if (window.innerWidth > 992) {
      document.documentElement.classList.toggle('nav-open');
    }
  };

  const parseChildrenRoutes = (children, i) => {
    const routes = [];
    for (let j = 0; j < children.length; j += 1) {
      const childRoute = children[j];
      routes.push(
        <Route
          path={childRoute.path}
          key={`${i}.${j}`}
          component={childRoute.component}
        />
      );
      if (
        typeof childRoute.children !== 'undefined' &&
        childRoute.children.length > 0
      ) {
        const childrenCRoutes = parseChildrenRoutes(
          childRoute.children,
          `${i}.${j}`
        );
        routes.push(childrenCRoutes);
      }
    }
    return routes;
  };

  const parseRoutes = () => {
    const routes = [];
    for (let i = 0; i < indexRoutes.length; i += 1) {
      const route = indexRoutes[i];
      let newRoute = [];
      if (route.component !== null) {
        newRoute = (
          <Route path={route.path} key={i} component={route.component} />
        );
      }
      if (route.name === 'Home') {
        newRoute = (
          <Route exact path={route.path} key={i} component={route.component} />
        );
      }
      routes.push(newRoute);
      if (typeof route.children !== 'undefined' && route.children.length > 0) {
        const childrenRoutes = parseChildrenRoutes(route.children, i);
        routes.push(childrenRoutes);
      }
    }
    const flattenedRoutes = [...routes];
    return flattenedRoutes;
  };

  const checkAppSession = useCallback(() => {
    const exec = async () => {
      const token = await new Promise((resolve) => {
        resolve(localStorage.getItem('token'));
      });
      const responseData = await axios({
        method: 'post',
        url: `${APIPath}admin-session`,
        crossDomain: true,
        data: { token },
      })
        .then((response) => response.data)
        .catch((error) => {
          console.log(error);
        });
      if (!responseData.status) {
        await dispatch(checkSession());
        setSessionRedirect(true);
      }
    };
    exec();
  }, [dispatch]);

  useEffect(() => {
    dispatch(loadSettings());
    dispatch(checkSession());
    dispatch(getLanguageCodes());
  }, [dispatch]);

  useEffect(() => {
    const interval = setInterval(() => {
      checkAppSession();
    }, 60000);
    return () => clearInterval(interval);
  }, [checkAppSession]);

  useEffect(() => {
    if (sessionActive) {
      openSidebar();
      loadProgressBar();
      dispatch(getSystemTypes());
      dispatch(getPeopleRoles());
      dispatch(getPersonTypes());
      dispatch(getOrganisationTypes());
      dispatch(getEventTypes());
      dispatch(loadDefaultEntities());
    }
  }, [sessionActive, dispatch]);

  useEffect(() => {
    if (loginRedirect) {
      dispatch(resetLoginRedirect());
    }
  }, [loginRedirect, dispatch]);

  useEffect(() => {
    if (seedRedirect) {
      dispatch(resetSeedRedirect());
    }
  }, [seedRedirect, dispatch]);

  useEffect(() => {
    if (sessionRedirect) {
      setSessionRedirect(false);
    }
  }, [sessionRedirect]);

  const routes = parseRoutes();
  const loginRedirectElem = loginRedirect ? <Redirect to="/" /> : [];
  const seedRedirectElem = seedRedirect ? <Redirect to="/" /> : [];
  const sessionRedirectElem = sessionRedirect ? <Redirect to="/login" /> : [];
  let appHTML = [];
  if (
    typeof settings.seedingAllowed !== 'undefined' &&
    settings.seedingAllowed
  ) {
    appHTML = (
      <Router basename={basename}>
        {seedRedirectElem}
        <Seed />
      </Router>
    );
  } else if (!sessionActive) {
    appHTML = <Login />;
  } else if (sessionActive) {
    appHTML = (
      <Router basename={basename}>
        <div className="wrapper" id="main-wrapper">
          {loginRedirectElem}
          {sessionRedirectElem}
          <Sidebar
            routes={indexRoutes}
            bgColor={backgroundColor}
            activeColor={activeColor}
          />
          <div className="main-panel" id="main-panel">
            <Header />
            <div className="content">
              <Switch>{routes}</Switch>
            </div>
            <Footer fluid extraClass="main-footer" />
          </div>
        </div>
      </Router>
    );
  }
  return <div>{appHTML}</div>;
}

export default App;
