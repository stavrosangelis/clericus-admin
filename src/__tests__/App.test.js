import React from 'react';
import { create } from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import App from '../App';

describe('renders app and tests init values', () => {
  it('render app', () => {
    const app = create(
      <Provider store={store()}>
        <App />
      </Provider>
    );
    const tree = app.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
