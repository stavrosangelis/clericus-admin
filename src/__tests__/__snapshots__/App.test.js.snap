// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app and tests init values render app 1`] = `<div />`;
